src/node/createSetupServer.ts

```diff
@@ -67,24 +67,28 @@ export function createSetupServer(
 
     interceptor.on('request', async function setupServerListener(request) {
       const mockedRequest = parseIsomorphicRequest(request)
-      const response = await handleRequest<MockedInterceptedResponse>(
-        mockedRequest,
-        currentHandlers,
-        resolvedOptions,
-        emitter,
-        {
-          transformResponse(response) {
-            return {
-              status: response.status,
-              statusText: response.statusText,
-              headers: response.headers.all(),
-              body: response.body,
-            }
-          },
+      const response = await handleRequest<
+        MockedInterceptedResponse & { delay?: number }
+      >(mockedRequest, currentHandlers, resolvedOptions, emitter, {
+        transformResponse(response) {
+          return {
+            status: response.status,
+            statusText: response.statusText,
+            headers: response.headers.all(),
+            body: response.body,
+            delay: response.delay,
+          }
         },
-      )
+      })
 
       if (response) {
+        await new Promise<void>((resolve) => {
+          if (!response.delay) {
```

Comment:
Since I've removed a hard-coded `setTimeout` around responses in `handleRequest`, the Node part now needs to delay mocked responses manually. The worker has been always delaying the responses in the worker thread so that the mocked delay counts for server computing time.